daemon/src/main/java/org/lsposed/lspd/service/LSPApplicationService.java

```diff
@@ -125,7 +126,20 @@ public ParcelFileDescriptor requestInjectedManagerBinder(List<IBinder> binder) t
             if (heartbeat != null) {
                 binder.add(ServiceManager.getManagerService().obtainManagerBinder(heartbeat, pid, uid));
             }
+        }
+        return ConfigManager.getInstance().getManagerApk();
+    }
 
+    @Override
+    public ParcelFileDescriptor requestCLIBinder(List<IBinder> binder) throws RemoteException {
+        ensureRegistered();
+        var pid = getCallingPid();
+        var uid = getCallingUid();
+        if (CLIValidator.managerStagePidValid(pid)) {
+            var heartbeat = handles.get(pid);
+            if (heartbeat != null) {
+                binder.add(ServiceManager.getManagerService().obtainManagerBinder(heartbeat, pid, uid));
```

Comment:
We need a CLIService instead of reusing ManagerService. It also benefits the pin code authorization.

Reply:
`cli -> BridgeService -> LSPApplicationService -> CLIService`
is it correct?
CLIService will implement almost all the methods of ManagerService.